core/src/port/mod.rs

```diff
@@ -292,15 +298,24 @@ impl Port {
         // turn on RSS
         if dev_info.flow_type_rss_offloads != 0 {
             port_conf.rxmode.mq_mode = dpdk::rte_eth_rx_mq_mode_ETH_MQ_RX_RSS;
-            port_conf.rx_adv_conf.rss_conf.rss_key = SYMMETRIC_RSS_KEY.as_ptr() as *mut u8;
-            port_conf.rx_adv_conf.rss_conf.rss_key_len = RSS_KEY_LEN as u8;
+            let symmetric_rss_key = if dev_info.hash_key_size == 52 {
+                SYMMETRIC_RSS_KEY_52.as_slice()
```

Comment:
Is it due to a Rust memory safeness thing that you can't use 40 bytes out of the 52 key length? Else, you can use simply the 52 one, just consider a length of 40 out of the 52 as they're the same.

Reply:
There was no reason, I unified the keys.